torch/csrc/autograd/variable.cpp

```diff
@@ -145,56 +80,49 @@ void Variable::backward(
 
   std::vector<Variable> inputs;
   if (!gradient.has_value()) {
-    gradient = make_variable(at::ones_like(data()), /*requires_grad=*/false);
+    gradient = make_variable(at::ones_like(*this), /*requires_grad=*/false);
```

Comment:
will this work if we enforce that the argument to `make_variable` doesn't have autograd meta?  Because the result of `at::ones_like` will already have autograd meta, right?  Isn't just calling `at::ones_like` enough here?